Yosemite/Yosemite/Stores/CardPresentPaymentStore.swift

```diff
@@ -110,6 +110,25 @@ private extension CardPresentPaymentStore {
             // https://stripe.dev/stripe-terminal-ios/docs/Enums/SCPPaymentIntentStatus.html#/c:@E@SCPPaymentIntentStatus@SCPPaymentIntentStatusRequiresCapture
             // TODO. Persist PaymentIntent, so that we can use it later to print a receipt
             // Deferred to https://github.com/woocommerce/woocommerce-ios/issues/3825
+
+            // Note: Here we transition from the Stripe Terminal Payment Intent
+            // to the WCPay backend managed Payment Intent, which we need
+            // to use to capture the payment.
+
+            self.remote.captureOrderPayment(for: siteID, orderID: orderID, paymentIntentID: intent.id) { result in
+                switch result {
+                case .success(let intent):
+                    if intent.status == .succeeded {
+                        onCompletion(.success(()))
+                    } else {
+                        DDLogDebug("Unexpected payment intent status \(intent.status) after attempting capture")
+                        onCompletion(.failure(CardReaderServiceError.paymentCapture()))
```

Comment:
@ctarda - or should CardPresentPaymentStore have its own Error enumeration here?

Reply:
What about passing the error you are passing now (paymentCapture) with a new case in the underlyingError (e.g. .remoteServiceError)? In any case, this looks like something that could be merged as is and be deferred to a subsequent ticket, what do you think?

Reply:
I don't know. I'm still thinking a separate enumeration for the payment store is more appropriate. I'll merge this and I've opened this ticket to explore further once we better understand the various ways the plugin and WCPay backend might fail to capture.